### DISCUSSION TITLE
Hide release-date option

### DISCUSSION THREAD

**Discussion by dev-yakuza** [stefanzweifel/php-changelog-updater#36] [Q&A] [answered]
Hi, there.

First, Thanks for making great open source.

Currently, the heading has the `release-date` like the following.

```
## 0.0.1 - 2023-05-21
```

But I want to hide the `release-date` like the following.

```
## 0.0.1
```

I've seen the `README` and the code.

- https://github.com/stefanzweifel/php-changelog-updater/blob/main/app/Commands/UpdateCommand.php#L48-L50

So, even if I pass the empty string to hide `release-date`, the current date is added.

Is there an option to hide the `release-date`? 🤔 If yes, please let me know it.

Thanks.

**Comment by stefanzweifel** [comment]
Hi @dev-yakuza 

No, there's currently no option to hide the `release-date`. 
Is there a specific reason, why you would like to hide the release date? Personally, I think the release date is quite an important information.

**Reply by dev-yakuza** [reply]
@stefanzweifel Thanks for the quick reply.

Here is the reason that I want it.

## 1. People can find the `release-date` in other ways

I use your GitHub Actions([changelog-updater-action](https://github.com/stefanzweifel/changelog-updater-action), [git-auto-commit-action](https://github.com/stefanzweifel/git-auto-commit-action)) to automate the Release note and changelog.

And I think the commit date of the `CHANGELOG.md` file is the same thing as the `release-date` if I use them.

<img width="1760" alt="Screenshot 2023-05-22 at 21 33 19" src="https://github.com/stefanzweifel/php-changelog-updater/assets/42969906/a81a6c8f-959e-4d1d-a051-d5d467b9817c">

- URL: https://github.com/dev-yakuza/bull/blame/main/CHANGELOG.md

And GitHub release has the date of the release date. It is also the same thing as the `release-date`.

<img width="1150" alt="Screenshot 2023-05-22 at 21 12 51" src="https://github.com/stefanzweifel/php-changelog-updater/assets/42969906/f618aef2-ddcf-4aba-94db-5ad0bdcaeea0">

So, If people use your GitHub Actions like me, other users can find it on Git history of the `CHANGELOG.md` file or release note on GitHub.

## 2. It looks not pretty on other platforms.

Recently, I try to make an open source on Flutter(https://pub.dev/)

It looks not pretty like the following.

| With release-date | Without release-date |
| -- | -- |
<img width="1753" alt="Screenshot 2023-05-22 at 21 17 33" src="https://github.com/stefanzweifel/php-changelog-updater/assets/42969906/87708c60-e5e9-45ba-b57d-e0d25068537b"> | <img width="1760" alt="Screenshot 2023-05-22 at 21 18 32" src="https://github.com/stefanzweifel/php-changelog-updater/assets/42969906/f26ac1c7-df71-48f4-90ba-0913b6f017dc"> |
| https://pub.dev/packages/bull/changelog | https://pub.dev/packages/build_runner/changelog |

**Reply by stefanzweifel** [reply]
@dev-yakuza I see. Didn't know about pub.dev and how they display versions.

I will make sure to add a `--hide-release-date` option to the CLI and Action. I'm a bit busy in the next couple of days, so might take a couple of days.

**Reply by stefanzweifel** [reply]
And for some background on the date. My goal or assumption is, that users are using the "Keep a changelog"-format for their changelogs. That format dictates a date.

Yes, the date could also be read from a GitHub release or even the commit, but having all that information in a central place like the changelog is a big plus in my opinion.

---

But I also see, that it can be a problem if third party platforms parse the changelog and it breaks some of their features. As my goal is to automate changelogs for everyone, I'm more than happy to add an option to make this possible.

**Reply by stefanzweifel** [reply, marked as answer]
I've added a `--hide-release-date` option to the CLI in https://github.com/stefanzweifel/php-changelog-updater/pull/39.
Will tag a new version in the next couple of days and update the Action as well.

**Reply by dev-yakuza** [reply]
@stefanzweifel Thank you so much ❤️

**Reply by stefanzweifel** [reply]
New release has been tagged for the CLI and the Action.

- https://github.com/stefanzweifel/changelog-updater-action/releases/tag/v1.8.0

You can now add a `hide-release-date` option to the Action.

```yml
      - name: Update Changelog
        uses: stefanzweifel/changelog-updater-action@v1
        with:
          latest-version: ${{ github.event.release.tag_name }}
          release-notes: ${{ github.event.release.body }}
          hide-release-date: true
```

**Reply by dev-yakuza** [reply]
@stefanzweifel Thank you so much 🙇
